### ISSUE TITLE
Option to suppress "Warning" and "Info" statements

### ISSUE THREAD

**Issue by ElOceanografo** [TuringLang/Turing.jl#1398]
Currently, sampling from a model always prints warnings for numerical errors and info statements on the initial step size, even with the option `verbose=false`.  This can really fill up the REPL, especially when a model run generates frequent numerical errors, or when fitting models to multiple datasets in parallel.  Would it be possible to optionally suppress these messages too?

**Comment by devmotion** [comment]
A cheap workaround would be to disable logging of such messages globally by setting
```julia
import Logging
Logging.disable_logging(Logging.Warn) # or e.g. Logging.Info
```
Alternatively, you can use a local logger for your `sample` call:
```julia
import Logging
logger = Logging.SimpleLogger(min_level=Logging.Error) # or e.g. Logging.Warn
chain = Logging.with_logger(logger) do
   sample(...)
end
```
The logging system is explained in more detail in [the Julia docs](https://docs.julialang.org/en/v1.5/stdlib/Logging).

A disadvantage of these approaches would be that progress bars are not displayed IIRC.

**Comment by cpfiffer** [comment]
This should probably be handled upstream in AdvancedHMC, since the warning doesn't respect `verbose`:

https://github.com/TuringLang/AdvancedHMC.jl/blob/63e32355bbc5be1639c9c12d21e908be0241db56/src/hamiltonian.jl#L47

**Comment by devmotion** [comment]
Yes, I agree. The approaches above are just workarounds as long as it is not fixed upstream.

**Comment by cpfiffer** [comment]
Yeah, it's a good thing to have the workaround on record.

**Comment by ElOceanografo** [comment]
Thanks, that workaround will be very helpful.  I'm fitting a large number of models, and while the first few "step size found" notifications are useful, they get less interesting after the 30 or 40 thousandth 🙃

**Comment by rikhuijzer** [comment]
The local logger code from David didn't work for me. Instead, on Julia 1.6, use
```
stream = IOBuffer(UInt8[])
logger = Logging.SimpleLogger(stream, Logging.Error)
chain = Logging.with_logger(logger) do
   sample(...)
end
```
In Julia 1.7, you can drop the `stream` argument thanks to https://github.com/JuliaLang/julia/pull/40423.

**Comment by andreasKroepelin** [comment]
This appears to be a serious issue when using Turing inside a Pluto notebook, as a large number of warnings makes the page unresponsive and I can't even hit the interrupt button.

**Comment by penelopeysm** [comment]
The NUTS 'initial step size' info message can now be suppressed with `sample(...; verbose=false)`. If there are other info messages that pop up please feel free to ping me and I'll be happy to disable them. _Warnings_ are a bit different; I'm kind of uneasy about suppressing warnings with `verbose=false`, but do feel free to comment if needed, happy to discuss.
